core/tests/puppeteer-acceptance-tests/puppeteer-testing-utilities/blogPostAdminUtils.js

```diff
@@ -0,0 +1,351 @@
+// Copyright 2023 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Blog Admin users utility file.
+ */
+
+const puppeteerUtilities = require('./puppeteer_utils.js');
+const testConstants = require(
+  '../puppeteer-testing-utilities/testConstants.js');
+const { showMessage } = require('./showMessageUtils.js');
+
+const blogTitleInput = 'input.e2e-test-blog-post-title-field';
+const blogBodyInput = 'div.e2e-test-rte';
+const thumbnailPhotoBox = 'e2e-test-photo-clickable';
+const unauthErrorContainer = 'div.e2e-test-error-container';
+const blogDashboardAuthorDetailsModal = 'div.modal-dialog';
+const roleUpdateUsernameInput = 'input#label-target-update-form-name';
+const blogEditorUsernameInput = 'input#label-target-form-reviewer-username';
+const maximumTagLimitInput = 'input#mat-input-0';
+const blogAuthorBioField = 'textarea.e2e-test-blog-author-bio-field';
+const blogDashboardUrl = testConstants.URLs.BlogDashboard;
+const blogAdminUrl = testConstants.URLs.BlogAdmin;
+
+module.exports = class e2eBlogPostAdmin extends puppeteerUtilities {
+  async addUserBioInBlogDashboard() {
+    await this.type(blogAuthorBioField, 'Dummy-User-Bio');
+    await this.page.waitForSelector(
+      'button.e2e-test-save-author-details-button:not([disabled])');
+    await this.clickOn('button', ' Save ');
+  }
+
+  async createDraftBlogPostWithTitle(draftBlogPostTitle) {
+    await this.addUserBioInBlogDashboard();
+    // See Note-1 below.
+    await this.page.waitForTimeout(500);
```

Comment:
No references in comments, please. Have the comment here. If you find yourself repeating comments then maybe the wait statement needs to be refactored into its own method with a comment, so that the comment appears just once.

I also don't understand the note. Why can't we then just wait for the dialog box to close (i.e. wait for an element to be removed from the screen)? Timeouts can lead to flakiness because different machines run at different speeds.

Reply:
that modal dialog box gets removed from the DOM instantly as we click the save button, but its the transition for which we have to wait for.. so to move further with the test.